t/plugin/clickhouse-logger.t

```diff
@@ -0,0 +1,194 @@
+            local ok, err = plugin.check_schema({
+                log_id = "syslog",
+                max_retry_count = 0,
+                retry_delay = 1,
+                buffer_duration = 60,
+                inactive_timeout = 10,
+                batch_max_size = 100,
+            })
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+value should match only one schema, but matches none
+
+
+
+=== TEST 4: add plugin on routes
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "clickhouse-logger": {
+                                "user": "default",
+                                "password": "a",
+                                "database": "default",
+                                "logtable": "t",
+                                "endpoint_addr": "http://127.0.0.1:8123"
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1982": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/opentracing"
+                }]],
+                [[{
+                    "node": {
+                        "value": {
+                            "plugins": {
+                                "clickhouse-logger": {
+                                    "user": "default",
+                                    "password": "a",
+                                    "database": "default",
+                                    "logtable": "t",
+                                    "endpoint_addr": "http://127.0.0.1:8123"
+                                }
+                            },
+                            "upstream": {
+                                "nodes": {
+                                    "127.0.0.1:1982": 1
+                                },
+                                "type": "roundrobin"
+                            },
+                            "uri": "/opentracing"
+                        },
+                        "key": "/apisix/routes/1"
+                    },
+                    "action": "set"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 5: access local server
+--- request
+GET /opentracing
+--- response_body
+opentracing
+--- error_log
+Batch Processor[http logger] successfully processed the entries
```

Comment:
We need to use a mock server to check the upload data, like https://github.com/apache/apisix/blob/5f6d16041f64402eb2af2c8380d532e6238c8717/t/plugin/loggly.t#L42-L61